packages/webapp/pages/squads/[handle]/[token].tsx

```diff
@@ -133,6 +141,11 @@ const SquadReferral = ({
   );
 
   const onJoinClick = async () => {
+    if (isBlockedUser) {
```

Comment:
We already have the current member's role, I don't think we need an additional state.
```suggestion
    if (member.source.currentMember.role === SquadMemberRole.Blocked) {
```